tests/unit/specs/scripts/common.spec.js

```diff
@@ -39,11 +39,13 @@ describe(`toMicroDenom`, () => {
 })
 
 describe(`sleep`, () => {
-  it(`returns a true as a promise`, () => {
-    jest.useFakeTimers()
-    sleep()
-    expect(setTimeout).toHaveBeenCalledTimes(1)
-    expect(sleep(10)).resolves.toEqual(10)
-    jest.useRealTimers()
+  it(`returns a true as a promise`, async () => {
+    const sleepCheck = () => {
+      return sleep(1000).then(() => {
+        return true
+      })
+    }
+    const res = await sleepCheck()
```

Comment:
you are now blocking all tests by 1s. if you use fakeTimers this test can pass in some milliseconds

Reply:
Yes... but this was one of the first things I tried:

```
    jest.useFakeTimers()
    const sleepCheck = () => {
      return sleep(1000).then(() => {
        return true
      })
    }
    const res = sleepCheck()
    jest.runAllTimers()
    expect(res).toBe(true)
    jest.useRealTimers()
```
And it surprisingly enough returns `{}` instead of `true`. Very unexpected :eyes: 

No undefined, no Promise... empty object :woman_shrugging:

Reply:
No, in the `then` block you need to set an external variable (`finished`)

Reply:
I am not really sure how you want me to do this... 

I have done this way:

```
  it(`the timer works properly, taking as many millisecons as we input and the Promise returns true`, done => {
    let check = false
    const cb = jest.fn(() => (check = true))

    jest.useFakeTimers()
    sleep(10000)
      .then(cb)
      .then(() => expect(cb).toHaveBeenCalled())
      .then(() => expect(check).toBe(true))
      .then(done)

    jest.runAllTimers()
    jest.useRealTimers()
  })
```

I think like this we are properly checking the function: both the timer and the resolved Promise.

Reply:
Looks good.

This didn't work?

```
it(`the timer works properly, taking as many millisecons as we input and the Promise returns true`, ()=> {
    let check = false

    jest.useFakeTimers()
    sleep(10000)
      .then(() => check = true)

    jest.runAllTimers()

    expect(check).toBe(true)

    jest.useRealTimers()
  })
```

Reply:
> Looks good.
> 
> This didn't work?
> 
> ```
> it(`the timer works properly, taking as many millisecons as we input and the Promise returns true`, ()=> {
>     let check = false
> 
>     jest.useFakeTimers()
>     sleep(10000)
>       .then(() => check = true)
> 
>     jest.runAllTimers()
> 
>     expect(check).toBe(true)
> 
>     jest.useRealTimers()
>   })
> ```

Yes, this is one of the first things I tried I was quite weirded out when I saw it didn't work...

But yeah, `check` remains `false` because this part of the code outside the `then` block (here concretely the `expect` line) will execute all before things inside `then` complete. 

It doesn't really matter that is written after the `jest.runAllTimers()` line. Everything outside the `then` block will be executed before.

So we both learned a lot here thanks to this not-so-boring-after-all test :wink: